src/parser/parser.rs

```diff
@@ -212,49 +122,75 @@ impl Parser {
 
     fn parse_expression(&mut self) -> Expression {
         let start = self.start_node();
-        let consumed_token = self.cur_token().clone();
-        self.bump_any();
-        let mut expr = match consumed_token.kind {
-            Kind::Identifier => Expression::Name(Box::new(Name {
+
+        let peeked_token = self.cur_token().clone();
+
+        let unary_expr = if is_unary_op(&peeked_token.kind) {
+            println!("Unary {:?}", peeked_token);
+            let op = map_unary_operator(&peeked_token.kind);
+            self.bump_any();
+            let operand = self.parse_expression();
+            let node = self.start_node();
+            Some(Expression::UnaryOp(Box::new(UnaryOperation {
+                node: self.finish_node(node),
+                op,
+                operand: Box::new(operand),
+            })))
+        } else {
+            None
+        };
+
+        // TODO fix
+        if unary_expr.is_none() {
+            self.bump_any();
+        }
+
+        println!("Now {:?}", peeked_token);
+        // refactor to if let and smaller function
```

Comment:
This part needs a refactor to not match on the token kind. It's not clear how do we expect these tokens and when should be bump the cursor.